shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/RouteCardData.kt

```diff
@@ -167,26 +167,28 @@ data class RouteCardData(
                     it.significance >= AlertSignificance.Secondary
             } ?: alertsDownstream.firstOrNull()
 
+        private data class PotentialService(val routeId: String, val headsign: String)
+
         /**
-         * Get all headsigns that might be shown for this leaf. For bus, the only headsigns that
-         * could be shown would be of the next two upcoming trips. For all other modes, headsigns
-         * for all of the upcoming trips **and** any other typical headsigns that are not reflected
-         * in the upcoming trips (may have already ended for the day, be disrupted, etc. but should
-         * still be considered) could be shown
+         * Get all routes and headsigns that might be shown for this leaf. For bus, the only
+         * headsigns that could be shown would be of the next two upcoming trips. For all other
+         * modes, headsigns for all of the upcoming trips **and** any other typical headsigns that
+         * are not reflected in the upcoming trips (may have already ended for the day, be
+         * disrupted, etc. but should still be considered) could be shown
          */
-        fun potentialHeadsigns(
+        private fun potentialService(
             now: Instant,
             representativeRoute: Route,
             globalData: GlobalResponse?,
-        ): Set<String> {
-            val potentialHeadsigns = mutableSetOf<String>()
+        ): Set<PotentialService> {
+            val potentialService = mutableSetOf<PotentialService>()
```

Comment:
Including the routes here should fix a bug that we may hit in Silver Line grouping: if the headsigns are all the same but the routes aren’t, we still need to treat the leaf as branched so that we can label the routes separately. It also makes picking the route in the no-trips case substantially easier.